feat(secure): add `:trust` command (#21107)
neovim/neovim: runtime/doc/editing.txt

   directory or in the directory of the current file (if the relative path
   starts with './' and 'd' is not included in 'cpoptions').

>    If Vim's cu rrent path is /u/user_x/work/release and you do >

Reviewer: ?

Author: Wow. I have no idea how that happened or how I missed that when checking the diff. Fixed in 24b35fe9d4b397e862178f8b8725b4631d41533d

   with depth limiter (/usr/**2) or upward search (;) notations.

 vim:tw=78:ts=8:noet:ft=help:norl:

==============================================================================

> 11. Secure Files						*secure*

Author: Done in d3d5654ca469c73ed196ac902a9e1eaadf4d0ec1

Secure Files are used to manage files that in a trust database, allowing

> trusted files to be sourced via |exrc|.

Author: >\`|vim.secure.read|\`
 I assume this should be `|vim.secure.read()|`?

> trusted files to be sourced via |exrc|.

Author: Done in d3d5654ca469c73ed196ac902a9e1eaadf4d0ec1
(with |vim.secure.read()| instead of  \`|vim.secure.read()|\`)

							*:trust* *E5570*
:trust [++deny] [++remove] [{file}]

> 			Manage file in the trust database. Without any options,
> 			`:trust` will mark file associated with current buffer
> 			as trusted.
> 			`++deny` will mark a file as denied and you won't be
> 			asked if you wan't to source that file.
> 			`++remove` will remove the file from the trust database.
> 			`++deny` and `++remove` accept optional parameter
> 			`{file}`, that is path to the file to be denied or
> 			removed.

Reviewer: Writing this in GitHub's review UI, so you will need to re-wrap it to fit within 79 cols.

```suggestion
			Manage files in the trust database. Without any options or arguments,
			`:trust` adds the file associated with the current buffer
			to the trust database, along with the SHA256 hash of its contents.
			`++deny` marks the file associated with the current buffer (or {file}, if given) as denied; no prompts will be displayed to the user and the file will never be executed.
			`++remove` removes the file associated with the current buffer (or {file}, if given) from the trust database. Future attempts to read the file in a secure setting (i.e. with 'exrc' or |vim.secure.read()|) will prompt the user if the file is trusted.
```

Author: Done in d3d5654ca469c73ed196ac902a9e1eaadf4d0ec1

   with depth limiter (/usr/**2) or upward search (;) notations.

 vim:tw=78:ts=8:noet:ft=help:norl:

> 

Author: Done in 74f9b054e9537c5ead30a61de0588037795ddf5d

==============================================================================
11. Trusted Files						*trust*

Nvim has the ability to execute arbitrary code through the 'exrc' option. In
order to prevent executing code from untrusted sources, Nvim has the concept of
"trusted files". An untrusted file will not be executed without the user's
consent, and a user can permanently mark a file as trusted or untrusted using
the |:trust| command or the |vim.secure.read()| function.

							*:trust* *E5570*
:trust [++deny] [++remove] [{file}]

			Manage files in the trust database. Without any options
			or arguments

> 			`:trust` adds the file associated with the current
> 			buffer to the trust database, along with the SHA256
> 			hash of its contents.
> 			`++deny` marks the file associated with the current
> 			buffer (or {file}, if given) as denied; no prompts will
> 			be displayed to the user and the file will never be
> 			executed.
> 			`++remove` removes the file associated with the current
> 			buffer (or {file}, if given) from the trust database.
> 			Future attempts to read the file in a secure setting

Reviewer: This is my mistake.

```suggestion
			:trust adds the file associated with the current
			buffer to the trust database, along with the SHA256
			hash of its contents.
			[++deny] marks the file associated with the current
			buffer (or {file}, if given) as denied; no prompts will
			be displayed to the user and the file will never be
			executed.
			[++remove] removes the file associated with the current
			buffer (or {file}, if given) from the trust database.
			Future attempts to read the file in a secure setting
```

Author: Done in 4b1d99b1cdd0ce92aa3867d0568d1c6e9a61e8df

> 			`:trust` adds the file associated with the current

Reviewer: ```suggestion
			:trust adds the file associated with the current
```

Author: Done in a87789fd1a4710209d3558f6314ea40713d43ddf
